### PR TITLE
use dc-polyfill instead of diagnostics_channel directly

### PR DESCRIPTION
**Pull request by tlhunter**
### What does this PR do?
- replaces calls to internal `diagnostics_channel` with `dc-polyfill`

### Motivation
- makes ESBuild plugin compatible with all Node.js versions that dd-trace is compatible with

### CODE REVIEW

**Review by Qard — APPROVED** [review]

File: yarn.lock
Comment (simon-id) [inline yarn.lock:1]: no addition to yarn.lock ? weird no ?

File: packages/dd-trace/src/appsec/iast/path-line.js [hunk 1]
```diff
@@ -13,11 +13,13 @@ const pathLine = {
 }
 
 const EXCLUDED_PATHS = [
-  path.join(path.sep, 'node_modules', 'diagnostics_channel')
+  path.join(path.sep, 'node_modules', 'dc-polyfill'),
+  path.join(path.sep, 'node_modules', 'dc-polyfill', 'dc-polyfill.js')
```
Comment (simon-id) [inline packages/dd-trace/src/appsec/iast/path-line.js:17]: @iunanua care to review just this bit ?

File: yarn.lock
Comment (tlhunter) [reply yarn.lock:1]: good point

Comment (tlhunter) [reply yarn.lock:1]: added

**Review by Qard — APPROVED** [review]

File: packages/dd-trace/src/appsec/iast/path-line.js (hunk 1 above)
Comment (iunanua) [inline packages/dd-trace/src/appsec/iast/path-line.js:17]: Why did you include `path.join(path.sep, 'node_modules', 'dc-polyfill', 'dc-polyfill.js')`? It should not be necessary

```suggestion
  path.join(path.sep, 'node_modules', 'dc-polyfill')
```

File: packages/dd-trace/src/appsec/iast/path-line.js [hunk 2]
```diff
@@ -13,11 +13,13 @@ const pathLine = {
...
 ]
 const EXCLUDED_PATH_PREFIXES = [
   'node:diagnostics_channel',
   'diagnostics_channel',
+  'dc-polyfill',
```
Comment (iunanua) [inline packages/dd-trace/src/appsec/iast/path-line.js:22]: Here native path requires only. So `dc-polyfill` is not needed.
```suggestion
```

**Review by iunanua — APPROVED** [review]

**Review by Qard — APPROVED** [review]

**Review by juan-fernandez — APPROVED** [review]

**Pull request merged**